File: templates/hexagonal-spring-boot/jpa-adapter/src/main/java/packageName/repository/ExampleRepository.java
package packageName.repository;

import java.util.List;
import java.util.stream.Collectors;
import packageName.domain.model.Example;
import packageName.domain.port.ObtainExample;
import packageName.repository.dao.ExampleDao;
import packageName.repository.entity.ExampleEntity;

public class ExampleRepository implements ObtainExample {

  private ExampleDao exampleDao;

  public ExampleRepository(ExampleDao exampleDao) {
    this.exampleDao = exampleDao;
  }

  @Override
  public List<Example> getAllExamples() {
    return exampleDao.findAll().stream().map(ExampleEntity::toModel).collect(Collectors.toList());
  }
}

File: templates/hexagonal-spring-boot-kotlin/jpa-adapter/src/main/kotlin/packageName/repository/ExampleRepository.kt
package packageName.repository

import packageName.domain.model.Example
import packageName.domain.port.ObtainExample
import packageName.repository.dao.ExampleDao

class ExampleRepository(private val exampleDao: ExampleDao) : ObtainExample {

    override fun getAllExamples(): List<Example> {
        return exampleDao.findAll().map { it.toModel() }
    }
}
